### ISSUE TITLE
react-native loaded into test runner

### ISSUE THREAD

**Issue by Brianzchen** [flow-typed/flow-typed#4534]
### CLI Version

4.x.x

### What problem do you want to solve?

As shown here https://github.com/flow-typed/flow-typed/pull/4533#discussion_r1355912499

There are cases where definitions want to use the types directly from react native just like we would with react. But since react-native is not part of the core lib this can't be possible

### Your take on the correct solution?

I think with a definitions `config.json` we can add a new property for third party libraries to install that are typed for the definition to depend upon during testing. 

We won't use this config during the install phase as we'll assume this is already catered for in a libraries package.json `dependencies` field. 

### Anything else?

Alternatively for `react-native` we could just load it into the test function regardless but I think that's not the right approach and would misleading

### Do you want to submit a pull request to implement this change?

Yes

**Comment by pascalduez** [comment]
Correct me if I'm wrong, but wasn't this the goal of the v4 / main branch? 
To be able to declare dependencies and have them loaded in tests.   

Also related or not https://flow.org/en/docs/react/multiplatform (experimental though).

**Comment by Brianzchen** [comment]
> okay got it, installing "real" dependencies, not just libdefs 😅

Yeah basically this. For most libraries core packages that have plugins/extensions aren't flow-typed in the first place so depending on their libdefs works well.

Right now I can only think of react-native being a use case. React itself also would be a usecase if it weren't bundled in the flow standard library.
